jerry-core/parser/js/js-parser-expr.c

```diff
@@ -1226,7 +1229,19 @@ parser_parse_unary_expression (parser_context_t *context_p, /**< context */
     }
     case LEXER_KEYW_THIS:
     {
-      parser_emit_cbc (context_p, CBC_PUSH_THIS);
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+      uint32_t constructor_this = PARSER_CLASS_HAS_SUPER | PARSER_CLASS_CONSTRUCTOR;
```

Comment:
I'm not sure whether a variable is needed for this, but if it is needed, at least make it const.

Reply:
Can be substituted via `PARSER_IS_CLASS_CONSTRUCTOR_SUPER` macro.